### ISSUE TITLE
Numeric tags are not spec compliant for Trans component substitution 

### ISSUE THREAD

**Issue by edkimmel** [i18next/react-i18next#1812]
## 🐛 Bug Report

https://github.com/formatjs/formatjs/issues/4488

tl;dr Numeric tags are not icu spec compliant and cause issues with the latest version of formatjs.  

We can swap to using the `{ [key]: component }` variation instead, but the array based method might want to be rethought or removed if it's found to not be compatible going forward.

## Your Environment

- *runtime version*: i.e. node v14, deno, browser xy
- *i18next version*: i.e. 19.5.3
- *os*: Mac, Windows, Linux
- *any other relevant information*

**Comment by adrai** [comment]
Since react-i18next is mainly used with the i18next format, and not first-class icu format, it's still ok, to use it...
But if you need to parse your resource texts and that raises issues, you really need to use the named component approach instead... 
Correct me if I'm wrong @jamuhl

**Comment by jamuhl** [comment]
agree. there are a lot of things not supported or working differentcwhen opting out of i18next format
